File: server/PlayItLoud.API/Models/Track.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PlayItLoud.API.Models
{
    [Table("Track")]
    public class Track
    {
        public int Id { get; private set; }

        [Required]
        [StringLength(1000)]
        public string Name { get; set; }

        [Range(1, int.MaxValue)]
        public int Order { get; set; }

        [Required]
        public TimeSpan Duration { get; set; }

        [Required]
        public int AlbumId { get; set; }

        public Album Album { get; set; }
    }
}

File: client/src/app/albums/models/track.ts
import { Album } from "./album";

export interface Track {
    id: number;
    name: string;
    order: number;
    duration?: string;
    albumId: number;
    album: Album;
}

export interface TrackDTO {
    name: string;
    order: number;
    duration?: string;
}